halo2_proofs/src/plonk/evaluation.rs

```diff
@@ -976,7 +975,11 @@ impl<C: CurveAffine> Evaluator<C> {
     }
 
     fn compute_cluster_idx(degree: usize, max_cluster_idx: usize) -> usize {
-        let mut idx = (31 - (degree as u32).leading_zeros()) as usize;
+        let mut idx = if degree != 0 {
+            (31 - (degree as u32).leading_zeros()) as usize
+        } else {
+            0
+        };
         if 1 << idx < degree {
             idx = idx + 1;
         }
```

Comment:
I suggest the following rewrite:

```rust
let mut idx = 0;

if degree != 0 {
    idx = degree.ilog2() as usize;

    if !degree.is_power_of_two() {
         idx += 1;
    }   
} 
   
```   
See https://godbolt.org/z/YEPd6n5cE

Reply:
ilog2() is unstable, use log2().floor() instead.